### ISSUE TITLE
Question about calculating MAdds of dynamic network in the paper

### ISSUE THREAD

**Issue by sseung0703** [changlin31/DS-Net#2]
Thank you for your great work, and I have a question about how to calculate MAdds in your paper.
The dynamic network has different widths and MAdds for each instance, but you denoted MAdds for your networks.
Are they the average MAdds for the whole dataset?

**Comment by changlin31** [comment]
Yes, the MAdds are averaged over the validation set, _e.g._:
https://github.com/changlin31/DS-Net/blob/2ae9455f7e7c2f3fe14b7e1a395811feabb208e3/dyn_slim/apis/train_slim_gate.py#L287

**Comment by sseung0703** [comment]
Thank you for your rapid answer. 😊
I think it is hard to know how much MAdds is required before the validation, but you achieved similar MAdds to comparative algorithms.
Is there any rule to achieve specific MAdds?

**Comment by changlin31** [comment]
Thanks for the valuable question.
The gate is actually very sensitive to hyperparameters. To avoid troublesome tuning of the loss balancing factors and other hyperparameters, we use different routing space for different network, _e.g._ we only use the slimmest few sub-networks to form the DS-Net-S.

This can be find in Appendix.A of the paper
<img width="513" alt="image" src="https://user-images.githubusercontent.com/61453811/115667297-3ffed380-a378-11eb-9c2d-33ed7eeaa2a7.png">

In practice, We first test the MAdds for each sub-networks, then manually choose those sub-networks with MAdds around the target MAdds as the dynamic routing space.
